text/0240-app-telemetry-schema-vision-roadmap.md

```diff
@@ -0,0 +1,429 @@
+group, app and lib developers, is much larger than the second and would not need
+to understand the details of the second format.
+
+### What should be the schema(s) be named?
+
+The naming of the schema(s) was also discussed but without consensus. The current proposals are as follows:
+
+- Single schema approach: Retain the existing Telemetry Schema name, supporting
+different formats depending on the use case (e.g., YAML for app and lib
+developers, JSON + JSON schema for publication and tool consumption).
+- Two schemas approach:
+  - Component Telemetry Schema alternative names: Telemetry Schema, Weaver Schema.
+  - Resolved Telemetry Schema alternative names: Compiled Telemetry Schema, Published Telemetry Schema, Weaver Schema.
+
+### What distribution mechanism should be used for retrieving dependency schemas?
+
+Currently, two main methods are being considered:
+
+1) Establishing a public, centralized repository for resolved schemas, where
+   library authors can register their resolved telemetry schemas as part of their
+   build process.
+2) Embedding the Resolved Telemetry Schema directly within the library artifact
+   (such as binary libraries, jar files, crates, Go modules, etc.), ensuring
+   automatic collection during the application's build process.
+
+At present, the second option is preferred as it is fully decentralized and
+eliminates the need for a global schema registry. A specific OTEP will be
+developed to define this distribution mechanism.
+
+## Roadmap
+
+### OTEPs
+
+To facilitate the review process and progressively deliver value to the
+OpenTelemetry community, a series of OTEPs and tools are suggested.
+
+* **Telemetry Schema(s) - Structure and Formats**: This OTEP will weigh the pros
+  and cons of a single schema versus a dual-schema approach. It aims to identify
+  the optimal solution and define the structures and formats for the two
+  concepts introduced in this OTEP: the Component and Resolved Telemetry Schema.
+  We plan several sub-OTEPs to progressively introduce and phase the concepts
+  discussed in this document:
+  * **Phase 1**: Introduce a registry section with a list of fully
+  resolved attributes, metrics, and other telemetry signals from the existing
+  semantic convention registry. Upon implementation, the OpenTelemetry project's
+  telemetry schema will include a registry of all standardized attributes,
+  metrics, and signals as defined by the semantic conventions.
+  * **Phase 2**: Add sections for resource, instrumentation library,
+  and schema sections to represent telemetry signals emitted by an application
+  (and its dependencies) or a library. Initially, only metrics, logs, and spans
+  definitions will be supported.
+  * **Phase 3**: Enable application or library authors to
+  override definitions inherited from a parent resolved schema.
+  * **Phase 4**: Implement support for events in the schema
+  section, pending the approval of events in OpenTelemetry.
```

Comment:
Is this included because you consider it a new signal type? It feels somewhat irrelevant to this project, a specific sub-domain that will simply follow the same rules as the other sub-domains.

Reply:
I have included events in a separate phase for the following reasons:
- It's a new type of signal that I believe has not yet been approved.
- If it is approved, and although it is protocol-compatible with log-type signals, it will be necessary to reify this concept at the telemetry schema level to make the most of it.